File: fix-component-names.js
#!/usr/bin/env node

import fs from 'fs';
import { glob } from 'glob';

// Function to fix component names that start with numbers
function fixComponentName(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    let modified = false;

    // Extract the directory name from the path
    const pathParts = filePath.split('/');
    const fileName = pathParts[pathParts.length - 2];
    
    // Convert kebab-case to PascalCase, handling numbers at the start
    let componentName = fileName.split('-').map(word => {
      // If word starts with a number, prefix with 'Ai' or 'Tech'
      if (/^\d/.test(word)) {
        return 'Tech' + word.charAt(0).toUpperCase() + word.slice(1);
      }
      return word.charAt(0).toUpperCase() + word.slice(1);
    }).join('');

    // Handle special cases
    if (componentName.startsWith('5g')) {
      componentName = 'FiveG' + componentName.slice(2);
    }
    if (componentName.startsWith('3d')) {
      componentName = 'ThreeD' + componentName.slice(2);
    }

    // Fix the component usage in the file
    const componentPattern = /<(\w+) \{\.\.\.props\} \/>/;
    const match = content.match(componentPattern);
    
    if (match) {
      const oldComponentName = match[1];
      if (oldComponentName !== componentName) {
        content = content.replace(new RegExp(`<${oldComponentName}`, 'g'), `<${componentName}`);
        modified = true;
      }
    }

    // Also fix the function name
    const functionPattern = /export default function (\w+)\(props: any\)/;
    const functionMatch = content.match(functionPattern);
    
    if (functionMatch) {
      const oldFunctionName = functionMatch[1];
      if (oldFunctionName !== componentName) {
        content = content.replace(new RegExp(`function ${oldFunctionName}`, 'g'), `function ${componentName}`);
        modified = true;
      }
    }

    if (modified) {
      fs.writeFileSync(filePath, content, 'utf8');
      console.log(`Fixed component name in: ${filePath} -> ${componentName}`);
      return true;
    }
    
    return false;
  } catch (error) {
    console.error(`Error fixing ${filePath}:`, error.message);
    return false;
  }
}

// Main execution
async function main() {
  console.log('Starting component name fixes...');

  // Find all page.tsx files
  const pageFiles = await glob('app/**/page.tsx', { cwd: process.cwd() });

  let fixedCount = 0;
  const totalCount = pageFiles.length;

  console.log(`Found ${totalCount} page files to process...`);

  pageFiles.forEach(file => {
    if (fixComponentName(file)) {
      fixedCount++;
    }
  });

  console.log(`\nFixed ${fixedCount} out of ${totalCount} files.`);
  console.log('Component name fixes completed!');
}

main().catch(console.error);

File: fix_component_names.py
#!/usr/bin/env python3
import os
import re
import glob

def fix_component_names():
    # Find all page.tsx files
    pattern = "app/**/page.tsx"
    files = glob.glob(pattern, recursive=True)
    
    fixed_count = 0
    
    for file_path in files:
        try:
            with open(file_path, 'r', encoding='utf-8') as f:
                content = f.read()
            
            original_content = content
            
            # Get the directory name from the file path
            dir_name = os.path.dirname(file_path).split('/')[-1]
            if not dir_name or dir_name == 'app':
                continue
            
            # Convert kebab-case to PascalCase for component name
            title = ''.join(word.capitalize() for word in dir_name.split('-'))
            
            # Fix component names that start with numbers
            if re.match(r'^\d', title):
                # Convert numbers to words
                number_map = {
                    '0': 'Zero', '1': 'One', '2': 'Two', '3': 'Three', '4': 'Four',
                    '5': 'Five', '6': 'Six', '7': 'Seven', '8': 'Eight', '9': 'Nine'
                }
                
                # Replace first character if it's a number
                if title[0] in number_map:
                    title = number_map[title[0]] + title[1:]
            
            # Fix the component declaration
            content = re.sub(
                r'const \w+Page: React\.FC = \(\) => \{',
                f'const {title}Page: React.FC = () => {{',
                content
            )
            
            # Fix the export statement
            content = re.sub(
                r'export default \w+Page',
                f'export default {title}Page',
                content
            )
            
            # Write the fixed content back if it changed
            if content != original_content:
                with open(file_path, 'w', encoding='utf-8') as f:
                    f.write(content)
                
                fixed_count += 1
                print(f"Fixed {file_path} -> {title}Page")
                
        except Exception as e:
            print(f"Error processing {file_path}: {e}")
    
    print(f"Fixed {fixed_count} files")

if __name__ == "__main__":
    fix_component_names()